src/redux/actions/layout/addWindow.js

```diff
@@ -38,9 +37,14 @@ const addWindow = (panel, window) => (dispatch, getState) => {
       return dispatch(updateLayout(newLayout.windows, panelExtra));
     }
 
-    newLayout.windows.first = getMultipleWindowsConfig(layout.windows.first, window || panel);
+    if (typeof newLayout.windows === 'string') {
```

Comment:
When theres only one or none windows left the structure is different and without this if the function would fail as assigning a property to undefined